Use article mainWebspace as default preview webspace

wachterjohannes opened the pull request:
Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | yes
| Deprecations? | no
| Fixed tickets | fixes none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

The preview now uses the article's mainWebspace setting as the default webspace for preview rendering. This fixes an issue where previewing an article in a locale not supported by the first webspace would fail with WebspaceLocalizationNotFoundException.

#### To Do

wachterjohannes committed:
Use article mainWebspace as default preview webspace

The preview now uses the article's mainWebspace setting as the default
webspace for preview rendering. This fixes an issue where previewing an
article in a locale not supported by the first webspace would fail with
WebspaceLocalizationNotFoundException.

Changes:
- Use formStore.data.mainWebspace as fallback in preview webspace selection
- Filter webspace options to only show webspaces supporting current locale
- Update webspace when locale changes to ensure compatibility

wachterjohannes committed:
Validate webspace locale support in data mappers

Validates that selected webspaces support the current locale at the data mapper
level. Throws InvalidArgumentException if a webspace doesn't support the locale,
preventing invalid configurations from being saved.

Changes:
- Added locale validation in WebspaceDataMapper for mainWebspace
- Added locale validation in AdditionalWebspacesDataMapper for all webspaces
- Throws exception if webspace doesn't support the locale (prevents save)
- Updated service definition to inject WebspaceManagerInterface
- Updated tests to mock webspace locale validation

wachterjohannes committed:
Add test cases for unsupported locale validation

Added test cases to verify that InvalidArgumentException is thrown when:
- mainWebspace doesn't support the current locale (WebspaceDataMapperTest)
- additionalWebspaces contain webspaces that don't support the locale (AdditionalWebspacesDataMapperTest)

Also added positive test cases to verify validation passes when locales are supported.

Prokyonn commented on packages/article/tests/Unit/Application/Content/DataMapper/AdditionalWebspacesDataMapperTest.php:
```diff
@@ -65,8 +75,17 @@ public function testMapCustomizeWebspaceSettingsTrue(): void
         $dimensionContent->setAdditionalWebspaces(['example-com'])->shouldBeCalled()->willReturn($dimensionContent->reveal());
 
+        // Mock webspace with locale support
+        $localization = new Localization('en');
```
```suggestion
        $localization = new Localization('en');
```

Prokyonn commented on packages/article/tests/Unit/Application/Content/DataMapper/AdditionalWebspacesDataMapperTest.php:
```diff
@@ -159,4 +178,62 @@ public function testMapWithEmptyAdditionalWebspaces(): void
+        $dimensionContent->getLocale()->willReturn('de');
+
+        // Mock webspace that only supports 'en'
+        $localizationEn = new Localization('en');
```
```suggestion
        $localizationEn = new Localization('en');
```

Prokyonn commented on packages/article/tests/Unit/Application/Content/DataMapper/AdditionalWebspacesDataMapperTest.php:
```diff
@@ -159,4 +178,62 @@ public function testMapWithEmptyAdditionalWebspaces(): void
+        $dimensionContent->setAdditionalWebspaces(['example-com'])->shouldBeCalled()->willReturn($dimensionContent->reveal());
+
+        // Mock webspace that supports 'en'
+        $localizationEn = new Localization('en');
```
```suggestion
        $localizationEn = new Localization('en');
```

Prokyonn commented on packages/content/tests/Unit/Content/Application/ContentDataMapper/DataMapper/WebspaceDataMapperTest.php:
```diff
@@ -130,4 +131,55 @@ public function testMapDataEmpty(): void
+        $localizedDimensionContent->setLocale('de');
+
+        // Mock webspace that only supports 'en'
+        $localizationEn = new Localization('en');
```
```suggestion
        $localizationEn = new Localization('en');
```

Prokyonn commented on packages/content/tests/Unit/Content/Application/ContentDataMapper/DataMapper/WebspaceDataMapperTest.php:
```diff
@@ -130,4 +131,55 @@ public function testMapDataEmpty(): void
+        $localizedDimensionContent->setLocale('en');
+
+        // Mock webspace that supports 'en'
+        $localizationEn = new Localization('en');
```
```suggestion
        $localizationEn = new Localization('en');
```

Prokyonn commented on src/Sulu/Bundle/PreviewBundle/Resources/js/containers/Preview/tests/Preview.test.js:
```diff
@@ -611,3 +612,33 @@ test('Change dateTime in PreviewStore when DatePicker changed', () => {
+    mount(<Preview formStore={formStore} router={router} />);
+
+    // Should use mainWebspace from formStore.data instead of first webspace
+    expect(PreviewStore).toBeCalledWith('articles', undefined, undefined, 'example', undefined);
```
```suggestion
    expect(PreviewStore).toBeCalledWith('articles', undefined, undefined, 'example', undefined);
```

Prokyonn commented on src/Sulu/Bundle/PreviewBundle/Resources/js/containers/Preview/tests/Preview.test.js:
```diff
@@ -611,3 +612,33 @@ test('Change dateTime in PreviewStore when DatePicker changed', () => {
+    mount(<Preview formStore={formStore} router={router} />);
+
+    // Should fall back to first webspace when mainWebspace is not available
+    expect(PreviewStore).toBeCalledWith('articles', undefined, undefined, 'sulu_io', undefined);
```
```suggestion
    expect(PreviewStore).toBeCalledWith('articles', undefined, undefined, 'sulu_io', undefined);
```

wachterjohannes committed:
Remove redundant comments from test files

Clean up test code by removing redundant comments that were flagged in code review.

wachterjohannes committed:
Fix ExampleControllerTest to use valid webspace

The test was using 'sulu-io2' which doesn't exist in the test configuration.
With the new webspace validation, this now throws an exception.
Changed to use 'sulu-io' which is the actual configured webspace.
